Marlin/src/module/endstops.cpp

```diff
@@ -422,20 +422,19 @@ void _O2 Endstops::M119() {
     print_es_state(READ(Z_MIN_PROBE_PIN) != Z_MIN_PROBE_ENDSTOP_INVERTING, PSTR(MSG_Z_PROBE));
   #endif
   #if ENABLED(FILAMENT_RUNOUT_SENSOR)
-    #define FRS_COUNT (1 + PIN_EXISTS(FIL_RUNOUT2) + PIN_EXISTS(FIL_RUNOUT3) + PIN_EXISTS(FIL_RUNOUT4) + PIN_EXISTS(FIL_RUNOUT5) + PIN_EXISTS(FIL_RUNOUT6))
-    #if FRS_COUNT == 1
-      print_es_state(READ(FIL_RUNOUT_PIN) != FIL_RUNOUT_INVERTING, MSG_FILAMENT_RUNOUT_SENSOR);
+    #if NUM_RUNOUT_SENSORS == 1
+      print_es_state((digitalRead(FIL_RUNOUT_PIN) != FIL_RUNOUT_INVERTING), PSTR(MSG_FILAMENT_RUNOUT_SENSOR));
     #else
       for (uint8_t i = 1; i <=
-        #if   FRS_COUNT == 6
+        #if   NUM_RUNOUT_SENSORS == 6
           6
-        #elif FRS_COUNT == 5
+        #elif NUM_RUNOUT_SENSORS == 5
           5
-        #elif FRS_COUNT == 4
+        #elif NUM_RUNOUT_SENSORS == 4
           4
-        #elif FRS_COUNT == 3
+        #elif NUM_RUNOUT_SENSORS == 3
           3
-        #elif FRS_COUNT == 2
+        #elif NUM_RUNOUT_SENSORS == 2
```

Comment:
It would be better to move the definition of `NUM_RUNOUT_SENSORS` to `Conditionals_post.h` and have it be automated, using the same method as shown above…
```cpp
#define NUM_RUNOUT_SENSORS (1 + PIN_EXISTS(FIL_RUNOUT2) + PIN_EXISTS(FIL_RUNOUT3) + PIN_EXISTS(FIL_RUNOUT4) + PIN_EXISTS(FIL_RUNOUT5) + PIN_EXISTS(FIL_RUNOUT6))
```
This would allow any number of filament runout sensors to be used on any combination of extruders, and remove the need for users to define `NUM_RUNOUT_SENSORS` with the assumption that they must always be on the first `NUM_RUNOUT_SENSORS` extruders.

It depends on whether we want to allow that kind of non-contiguous setup.